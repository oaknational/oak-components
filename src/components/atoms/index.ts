export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
