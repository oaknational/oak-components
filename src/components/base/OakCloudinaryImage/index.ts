export * from "./OakCloudinaryImage";
export { setCloudinaryConfig } from "./cloudinary";
