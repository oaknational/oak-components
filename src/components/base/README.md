# Base components

## Rules for components in this folder

- Imports from other base components
- Imports from Ui and Integrated components are forbidden!
- Generic
- Unstyled
- Avoid embedded logic and state
- No sub-components
  (eg. Box)
