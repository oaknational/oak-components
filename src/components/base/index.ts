export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakBaseCard";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
