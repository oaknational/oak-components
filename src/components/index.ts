export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
