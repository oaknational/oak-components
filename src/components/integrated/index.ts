export * from "./OakQuizCheckBox";
export * from "./OakQuizTextInput";
export * from "./OakQuizRadioButton";
export * from "./OakQuizHint";
export * from "./OakQuizBottomNav";
export * from "./OakLessonTopNav";
export * from "./OakQuizFeedback";
export * from "./OakQuizCounter";
