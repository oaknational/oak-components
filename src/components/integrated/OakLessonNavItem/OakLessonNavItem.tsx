import React, { ComponentPropsWithoutRef, ElementType } from "react";
import styled, { css } from "styled-components";

import {
  OakBox,
  OakFlex,
  OakIcon,
  OakIconName,
  OakSpan,
} from "@/components/base";
import { parseColor } from "@/styles/helpers/parseColor";
import { OakRoundIcon } from "@/components/ui";
import { parseColorFilter } from "@/styles/helpers/parseColorFilter";
import { parseSpacing } from "@/styles/helpers/parseSpacing";
import { parseDropShadow } from "@/styles/helpers/parseDropShadow";
import { OakCombinedColorToken } from "@/styles";

type LessonSectionName = "intro" | "starter-quiz" | "video" | "exit-quiz";

type BaseOakLessonNavItemProps<C extends ElementType> = {
  as?: C;
  /**
   * Disable the section preventing navigation to it.
   */
  disabled?: boolean;
} & ComponentPropsWithoutRef<C>;

type QuizSectionProps = {
  lessonSectionName: "starter-quiz" | "exit-quiz";
  /**
   * The number of questions in the quiz
   */
  numQuestions: number;
  /**
   * The number of questions answered correctly
   */
  grade: number;
};

type VideoSectionProps = {
  lessonSectionName: "video";
  /**
   * The length of the video in minutes
   */
  videoLength: number;
};

type IntroSectionProps = {
  lessonSectionName: "intro";
};

type SectionProps = {
  /**
   * Denotes the progress in the lesson section
   */
  progress: "not-started" | "in-progress" | "complete";
} & (IntroSectionProps | QuizSectionProps | VideoSectionProps);

export type OakLessonNavItemProps<C extends ElementType> =
  BaseOakLessonNavItemProps<C> & SectionProps;

const StyledLabel = styled(OakBox)``;

const StyledRoundIcon = styled(OakRoundIcon)<{
  $disabled?: boolean;
}>`
  width: ${parseSpacing("all-spacing-8")};
  height: ${parseSpacing("all-spacing-8")};
  padding: 0;

  background: transparent;

  img {
    filter: ${(props) =>
      parseColorFilter(props.$disabled ? "icon-disabled" : "icon-inverted")};
  }
`;

const activeIconStyles = css`
  ${StyledRoundIcon} {
    background: ${parseColor("bg-btn-primary")};

    img {
      filter: ${parseColorFilter("icon-main")};
    }
  }
`;

const StyledLessonNavItem = styled(OakFlex)<{ $disabled?: boolean }>`
  outline: none;
  text-align: initial;

  &:focus-visible {
    box-shadow: ${parseDropShadow("drop-shadow-centered-lemon")},
      ${parseDropShadow("drop-shadow-centered-grey")};
  }

  ${(props) => props.$disabled && "cursor: default"}

  ${(props) =>
    !props.$disabled &&
    css`
      cursor: pointer;

      /* Don't apply hover styles on touch devices */
      @media (hover: hover) {
        &:hover {
          ${StyledLabel} {
            text-decoration: underline;
          }

          ${activeIconStyles}
        }
      }

      &:active {
        box-shadow: ${parseDropShadow("drop-shadow-lemon")},
          ${parseDropShadow("drop-shadow-grey")};
        ${activeIconStyles}
      }
    `}
`;

const FlexedOakBox = styled(OakBox)`
  flex: 1;
`;

/**
 * Enables navigation to the given section of a lesson as well as displaying current progress
 */
export const OakLessonNavItem = <C extends ElementType = "a">(
  props: OakLessonNavItemProps<C>,
) => {
  const { as, lessonSectionName, progress, disabled, href, onClick, ...rest } =
    props;
  const [notStartedBackgroundColor, backgroundColor, borderColor] =
    pickColorsForSection(lessonSectionName);

  return (
    <StyledLessonNavItem
      as={disabled ? "div" : as ?? "a"}
      $gap="space-between-m"
      $alignItems="center"
      $background={
        progress === "not-started" ? notStartedBackgroundColor : backgroundColor
      }
      $ph={["inner-padding-m", "inner-padding-xl"]}
      $pv="inner-padding-l"
      $borderRadius="border-radius-l"
      $borderColor={borderColor}
      $ba="border-solid-l"
      $disabled={disabled}
      $color="text-primary"
      href={disabled ? undefined : href}
      onClick={disabled ? undefined : onClick}
      {...rest}
    >
      <OakFlex $width="all-spacing-13" $justifyContent="center">
        <OakIcon
          iconName={pickIconForSection(lessonSectionName)}
          $width="all-spacing-10"
          $height="all-spacing-10"
        />
      </OakFlex>
      <FlexedOakBox>
        <StyledLabel
          as="strong"
          $font={["heading-6", "heading-5"]}
          $color={disabled ? "text-disabled" : "text-primary"}
        >
          {pickLabelForSection(lessonSectionName)}
        </StyledLabel>
        <OakBox $font={["body-2", "body-1"]}>
          {pickSummaryForProgress(props)}
        </OakBox>
      </FlexedOakBox>
      {renderQuestionCounter(props)}
      <StyledRoundIcon iconName="chevron-right" $disabled={disabled} />
    </StyledLessonNavItem>
  );
};

function renderQuestionCounter(props: SectionProps) {
  if (props.progress !== "complete") {
    return null;
  }

  /**
   * The large answer counter is only rendered when on a non-mobile screen
   */
  switch (props.lessonSectionName) {
    case "exit-quiz":
    case "starter-quiz":
      return (
        <OakBox $display={["none", "block"]} $mr="space-between-m">
          <OakSpan $font="heading-4">{props.grade}</OakSpan>
          <OakSpan $font="heading-6">&nbsp;/&nbsp;{props.numQuestions}</OakSpan>
        </OakBox>
      );
    default:
      return null;
  }
}

function pickIconForSection(sectionName: LessonSectionName): OakIconName {
  switch (sectionName) {
    case "intro":
      return "intro";
    case "starter-quiz":
    case "exit-quiz":
      return "quiz";
    case "video":
      return "video";
  }
}

function pickColorsForSection(
  sectionName: LessonSectionName,
): [
  notStartedBackgroundColor: OakCombinedColorToken,
  backgroundColor: OakCombinedColorToken,
  borderColor: OakCombinedColorToken,
] {
  switch (sectionName) {
    case "intro":
      return [
        "bg-decorative2-very-subdued",
        "bg-decorative2-main",
        "border-decorative2-stronger",
      ];
    case "starter-quiz":
      return [
        "bg-decorative1-very-subdued",
        "bg-decorative1-main",
        "border-decorative1-stronger",
      ];
    case "video":
      return [
        "bg-decorative4-very-subdued",
        "bg-decorative4-main",
        "border-decorative4-stronger",
      ];
    case "exit-quiz":
      return [
        "bg-decorative5-very-subdued",
        "bg-decorative5-main",
        "border-decorative5-stronger",
      ];
  }
}

function pickLabelForSection(sectionName: LessonSectionName): string {
  switch (sectionName) {
    case "intro":
      return "Intro";
    case "starter-quiz":
      return "Starter quiz";
    case "video":
      return "Watch video";
    case "exit-quiz":
      return "Exit quiz";
  }
}

function pickSummaryForProgress(props: SectionProps) {
  switch (props.progress) {
    case "not-started":
      return pickSummaryForNotStarted(props);
    case "in-progress":
      return "In progress...";
    case "complete":
      return (
        <OakFlex $gap="space-between-sssx" $alignItems="center">
          <OakIcon
            iconName="tick"
            $width="all-spacing-6"
            $height="all-spacing-6"
          />
          {pickSummaryForComplete(props)}
        </OakFlex>
      );
  }
}

function pickSummaryForNotStarted(props: SectionProps) {
  switch (props.lessonSectionName) {
    case "intro":
      return "Get ready";
    case "starter-quiz":
      return `${props.numQuestions} Questions`;
    case "exit-quiz":
      return `Practice ${props.numQuestions} questions`;
    case "video":
      return `${props.videoLength} min`;
  }
}

function pickSummaryForComplete(props: SectionProps) {
  switch (props.lessonSectionName) {
    case "intro":
    case "video":
      return "Completed";
    case "starter-quiz":
    case "exit-quiz":
      // The counter is rendered as the summary next
      // when on a mobile device, so it is hidden on larger screens
      return (
        <>
          <OakBox $display={["none", "block"]}>Completed</OakBox>
          <OakBox $display={["block", "none"]}>
            {props.grade}/{props.numQuestions} correct
          </OakBox>
        </>
      );
  }
}
