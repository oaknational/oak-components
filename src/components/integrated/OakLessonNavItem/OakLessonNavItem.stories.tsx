import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakLessonNavItem } from "./OakLessonNavItem";

import { OakFlex } from "@/components/base";

const meta: Meta<typeof OakLessonNavItem> = {
  component: OakLessonNavItem,
  tags: ["autodocs"],
  title: "components/integrated/OakLessonNavItem",
  argTypes: {
    numQuestions: {
      control: {
        type: "number",
      },
    },
    grade: {
      control: {
        type: "number",
      },
    },
    videoLength: {
      control: {
        type: "number",
      },
      if: { arg: "lessonSectionName", eq: "video" },
    },
  },
  args: {
    lessonSectionName: "intro",
    progress: "not-started",
    href: "#",
    answerCount: 4,
    questionCount: 6,
    videoLength: 20,
  },
  decorators: [
    (Story) => {
      return (
        <OakFlex $flexDirection="column" $gap="space-between-m">
          {Story()}
        </OakFlex>
      );
    },
  ],
  parameters: {
    controls: {
      include: [
        "progress",
        "lessonSectionName",
        "isDisabled",
        "answerCount",
        "questionCount",
        "videoLength",
      ],
    },
  },
};
export default meta;

type Story = StoryObj<typeof OakLessonNavItem>;

export const Default: Story = {
  render: (args) => <OakLessonNavItem {...args} />,
};

export const AsAButton: Story = {
  render: (args) => <OakLessonNavItem {...args} as="button" />,
};

export const NotStarted: Story = {
  render: () => (
    <>
      <OakLessonNavItem lessonSectionName="intro" progress="not-started" />
      <OakLessonNavItem
        lessonSectionName="starter-quiz"
        progress="not-started"
        questionCount={6}
        answerCount={0}
      />
      <OakLessonNavItem
        lessonSectionName="video"
        progress="not-started"
        videoLength={20}
      />
      <OakLessonNavItem
        lessonSectionName="exit-quiz"
        progress="not-started"
        questionCount={4}
        answerCount={0}
      />
    </>
  ),
};

export const InProgress: Story = {
  render: () => (
    <>
      <OakLessonNavItem lessonSectionName="intro" progress="in-progress" />
      <OakLessonNavItem
        lessonSectionName="starter-quiz"
        progress="in-progress"
        questionCount={6}
        answerCount={0}
      />
      <OakLessonNavItem
        lessonSectionName="video"
        progress="in-progress"
        videoLength={20}
      />
      <OakLessonNavItem
        lessonSectionName="exit-quiz"
        progress="in-progress"
        questionCount={6}
        answerCount={0}
      />
    </>
  ),
};

export const Complete: Story = {
  render: () => (
    <>
      <OakLessonNavItem lessonSectionName="intro" progress="complete" />
      <OakLessonNavItem
        lessonSectionName="starter-quiz"
        progress="complete"
        questionCount={6}
        answerCount={5}
      />
      <OakLessonNavItem
        lessonSectionName="video"
        progress="complete"
        videoLength={20}
      />
      <OakLessonNavItem
        lessonSectionName="exit-quiz"
        progress="complete"
        questionCount={4}
        answerCount={4}
      />
    </>
  ),
};
