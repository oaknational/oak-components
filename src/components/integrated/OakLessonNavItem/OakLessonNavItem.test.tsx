import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";

import { OakLessonNavItem } from "./OakLessonNavItem";

import { OakThemeProvider } from "@/components/base";
import { oakDefaultTheme } from "@/styles";
import renderWithTheme from "@/test-helpers/renderWithTheme";

describe(OakLessonNavItem, () => {
  it("matches snapshot", () => {
    const tree = create(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakLessonNavItem lessonSectionName="intro" progress="not-started" />,
      </OakThemeProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders copy for each lesson section that has not been started", () => {
    const { getByTestId } = renderWithTheme(
      <>
        <OakLessonNavItem
          data-testid="intro"
          lessonSectionName="intro"
          progress="not-started"
        />
        <OakLessonNavItem
          data-testid="starter-quiz"
          lessonSectionName="starter-quiz"
          progress="not-started"
          numQuestions={6}
          grade={0}
        />
        <OakLessonNavItem
          data-testid="video"
          lessonSectionName="video"
          progress="not-started"
          videoLength={20}
        />
        <OakLessonNavItem
          data-testid="exit-quiz"
          lessonSectionName="exit-quiz"
          progress="not-started"
          numQuestions={6}
          grade={5}
        />
      </>,
    );

    expect(getByTestId("intro").textContent).toContain("Get ready");
    expect(getByTestId("starter-quiz").textContent).toContain("6 Questions");
    expect(getByTestId("exit-quiz").textContent).toContain(
      "Practice 6 questions",
    );
    expect(getByTestId("video").textContent).toContain("20 min");
  });

  it("renders copy for each lesson section that has not been started", () => {
    const { getByTestId } = renderWithTheme(
      <>
        <OakLessonNavItem
          data-testid="intro"
          lessonSectionName="intro"
          progress="in-progress"
        />
        <OakLessonNavItem
          data-testid="starter-quiz"
          lessonSectionName="starter-quiz"
          progress="in-progress"
          numQuestions={6}
          grade={0}
        />
        <OakLessonNavItem
          data-testid="video"
          lessonSectionName="video"
          progress="in-progress"
          videoLength={20}
        />
        <OakLessonNavItem
          data-testid="exit-quiz"
          lessonSectionName="exit-quiz"
          progress="in-progress"
          numQuestions={6}
          grade={5}
        />
      </>,
    );

    expect(getByTestId("intro").textContent).toContain("In progress...");
    expect(getByTestId("starter-quiz").textContent).toContain("In progress...");
    expect(getByTestId("exit-quiz").textContent).toContain("In progress...");
    expect(getByTestId("video").textContent).toContain("In progress...");
  });

  it("renders copy for each lesson section that has not been started", () => {
    const { getByTestId } = renderWithTheme(
      <>
        <OakLessonNavItem
          data-testid="intro"
          lessonSectionName="intro"
          progress="complete"
        />
        <OakLessonNavItem
          data-testid="starter-quiz"
          lessonSectionName="starter-quiz"
          progress="complete"
          numQuestions={6}
          grade={0}
        />
        <OakLessonNavItem
          data-testid="video"
          lessonSectionName="video"
          progress="complete"
          videoLength={20}
        />
        <OakLessonNavItem
          data-testid="exit-quiz"
          lessonSectionName="exit-quiz"
          progress="complete"
          numQuestions={6}
          grade={5}
        />
      </>,
    );

    expect(getByTestId("intro").textContent).toContain("Completed");
    expect(getByTestId("starter-quiz").textContent).toContain("Completed");
    expect(getByTestId("exit-quiz").textContent).toContain("Completed");
    expect(getByTestId("video").textContent).toContain("Completed");
  });
});
