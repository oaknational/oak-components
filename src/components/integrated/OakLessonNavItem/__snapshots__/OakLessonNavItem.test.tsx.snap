// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakLessonNavItem matches snapshot 1`] = `
[
  .c0 {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 1.25rem;
  padding-bottom: 1.25rem;
  background: #e7f6f5;
  border: 0.188rem solid;
  border-color: #7cd8d0;
  border-radius: 1rem;
  font-family: Lexend,sans-serif;
}

.c3 {
  width: 5rem;
  font-family: Lexend,sans-serif;
}

.c5 {
  position: relative;
  width: 3.5rem;
  height: 3.5rem;
  font-family: Lexend,sans-serif;
}

.c7 {
  font-family: Lexend,sans-serif;
}

.c9 {
  color: #222222;
  font-family: Lexend,sans-serif;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c10 {
  font-family: Lexend,sans-serif;
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c11 {
  width: 3rem;
  height: 3rem;
  padding: 0.25rem;
  background: #ffe555;
  border-radius: 6.25rem;
  font-family: Lexend,sans-serif;
}

.c14 {
  position: relative;
  width: 100%;
  height: 100%;
  font-family: Lexend,sans-serif;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  object-fit: contain;
}

.c13 {
  width: 2.5rem;
  height: 2.5rem;
  padding: 0;
  background: transparent;
}

.c13 img {
  -webkit-filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
}

.c2 {
  outline: none;
  text-align: initial;
  cursor: pointer;
}

.c2:focus-visible {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c2:hover .c12,
.c2:active .c12 {
  background: #222222;
}

.c2:hover .c12 img,
.c2:active .c12 img {
  -webkit-filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
}

.c2:active {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%), 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:750px) {
  .c0 {
    padding-left: 1.5rem;
  }
}

@media (min-width:750px) {
  .c0 {
    padding-right: 1.5rem;
  }
}

@media (min-width:750px) {
  .c9 {
    font-weight: 600;
  }
}

@media (min-width:750px) {
  .c9 {
    font-size: 1.5rem;
  }
}

@media (min-width:750px) {
  .c9 {
    line-height: 2rem;
  }
}

@media (min-width:750px) {
  .c9 {
    -webkit-letter-spacing: 0.0115rem;
    -moz-letter-spacing: 0.0115rem;
    -ms-letter-spacing: 0.0115rem;
    letter-spacing: 0.0115rem;
  }
}

@media (min-width:750px) {
  .c10 {
    font-weight: 300;
  }
}

@media (min-width:750px) {
  .c10 {
    font-size: 1.125rem;
  }
}

@media (min-width:750px) {
  .c10 {
    line-height: 1.75rem;
  }
}

@media (min-width:750px) {
  .c10 {
    -webkit-letter-spacing: -0.005rem;
    -moz-letter-spacing: -0.005rem;
    -ms-letter-spacing: -0.005rem;
    letter-spacing: -0.005rem;
  }
}

<a
    className="c0 c1 c2"
  >
    <div
      className="c3 c4"
    >
      <div
        className="c5"
      >
        <img
          alt="intro"
          className="c6"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="https://undefined/undefined/v1705662092/icons/pl7bnmb13txese9yxkjv.svg"
          style={
            {
              "bottom": 0,
              "color": "transparent",
              "height": "100%",
              "left": 0,
              "objectFit": undefined,
              "objectPosition": undefined,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="c7 c8"
    >
      <strong
        className="c9"
      >
        Intro
      </strong>
      <div
        className="c10"
      >
        Get ready
      </div>
    </div>
    <div
      className="c11 c12 c13"
    >
      <div
        className="c14"
      >
        <img
          alt="chevron-right"
          className="c6"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="https://undefined/undefined/v1699895475/icons/rhzft6wnpdp3nvsmlyei.svg"
          style={
            {
              "bottom": 0,
              "color": "transparent",
              "height": "100%",
              "left": 0,
              "objectFit": undefined,
              "objectPosition": undefined,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </a>,
  ",",
]
`;
